cmd/piper.go

```diff
@@ -162,6 +164,65 @@ func PrepareConfig(cmd *cobra.Command, metadata *config.StepData, stepName strin
 	return nil
 }
 
+func convertTypes(config map[string]interface{}, options interface{}) map[string]interface{} {
+	typedOptions := reflect.ValueOf(options)
+	if typedOptions.Kind() == reflect.Ptr {
+		typedOptions = typedOptions.Elem()
+	}
+	optionsType := typedOptions.Type()
+
+	for key := range config {
+		// Find the field in options which will be mapped to 'key'
```

Comment:
This is more a what comment than a why. Could we improve the method names or express it somehow in the code?